### DISCUSSION TITLE
Server IP - where to get this information from?

### DISCUSSION THREAD

**Discussion by M-Stenzel** [strongswan/strongswan#2524] [Q&A]
Hi team, 
I installed strongswan, version 5.9.12 on Linux.

This is the ipsec.conf file

```
config setup
    charondebug="ike 1, knl 1, cfg 0"
    uniqueids=no

conn ikev2-vpn
    auto=add
    compress=no
    type=tunnel
    keyexchange=ikev2
    fragmentation=yes
    forceencaps=yes
    dpdaction=clear
    dpddelay=300s
    rekey=no
    left=%any
    leftid=redacted_server_IP_address
    leftcert=/etc/pki/certs/server-cert.pem
    leftsendcert=always
    leftsubnet=0.0.0.0/0
    right=%any
    rightid=%any
    rightauth=eap-mschapv2
    rightsourceip=10.12.0.0/24
    rightsendcert=never
    eap_identity=%identity
    ike=aes256gcm16-sha512-prfsha512-ecp512bp
    esp=aes256gcm16-sha512-prfsha512-ecp512bp
```

In my Android client app I find this:

```
Nov  3 17:44:38 07[IKE] IKE_SA android[3] established between redacted[redacted]...IP[IP]
Nov  3 17:44:38 07[IKE] scheduling rekeying in 35959s
Nov  3 17:44:38 07[IKE] maximum IKE_SA lifetime 37759s
Nov  3 17:44:38 07[CFG] selected proposal: ESP:AES_GCM_16_256/NO_EXT_SEQ
Nov  3 17:44:38 07[IKE] CHILD_SA android{4} established with SPIs ef678c15_i c2a808bf_o and TS 10.12.0.2/32 === 0.0.0.0/0
Nov  3 17:44:38 07[DMN] setting up TUN device for CHILD_SA android{4}
Nov  3 17:44:38 07[DMN] successfully created TUN device
```

The client IP address is now 10.12.0.2. **But what is the server IP address?** How to get to know this?
I need that server IP to make owntracks using the VPN.

I could not find this information (cannot believe this...) in the documentation.

**Comment by tobiasbrunner** [comment]
> **But what is the server IP address?**

Any IP address it has assigned.

> How to get to know this?

`ip addr` perhaps?

If you want it to have a specific IP (e.g. from the address pool), you can assign one to e.g. `lo`.

**Reply by M-Stenzel** [reply]
Ah,
well, when I check with "ip addr" on the SERVER the 10.12.0.0/24 net (see above) is not shown.

One other try...

netstat -tulpn | grep charon
udp     3328      0 0.0.0.0:68              0.0.0.0:*                           23824/charon-system 
udp     3328      0 0.0.0.0:68              0.0.0.0:*                           18470/charon        
udp        0      0 0.0.0.0:500             0.0.0.0:*                           18470/charon        
udp        0      0 0.0.0.0:4500            0.0.0.0:*                           18470/charon        
udp6       0      0 :::500                  :::*                                18470/charon        
udp6       0      0 :::4500                 :::*                                18470/charon

**Reply by tobiasbrunner** [reply]
> well, wenn I check with "ip addr" on the SERVER the 10.12.0.0/24 net (see above) is not shown.

Why would there be (unless you installed such an IP address yourself)? The address pool you configured is for clients connecting to the server. If you want the server to have an IP from that subnet (might not be necessary, depending on the client), you have to assign it yourself (e.g. use `ip addr add 10.12.0.1/24 dev lo` and then change the pool to `rightsourceip=10.12.0.2/24` so that this address is not also assigned to a client).

**Reply by M-Stenzel** [reply]
That was exactly the information, that I needed. I configured as you suggested, and now everything works.
Now I finally can replace openvpn :)

Thank you again!
